resolver/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+//! TODO
+
+use std::future::Future;
+
+use bytes::Bytes;
+use commonware_utils::Serialize;
+use futures::channel::oneshot;
+use std::fmt::Debug;
+
+pub mod p2p;
+
+/// Used as the key to fetch data from the network.
+pub trait Key: Serialize + Clone + Debug + PartialEq + Eq + Ord + Send + Sync + 'static {}
```

Comment:
Not sure if I actually need this type. I could just use `Bytes` and have the application deal with all the serialization/deserialization...

Reply:
It's in a bit of a weird middle-state right now. Either I should define a `Value` type (similar to `Key`) that replaces the use of `Bytes` in this file, or I should just use `Bytes` everywhere in this file (and delete the `Key` type)